src/main/java/com/faforever/client/vault/replay/WatchButtonController.java

```diff
@@ -1,81 +1,132 @@
 import com.faforever.client.replay.ReplayService;
 import com.faforever.client.util.TimeService;
 import com.google.common.annotations.VisibleForTesting;
 import javafx.animation.KeyFrame;
 import javafx.animation.Timeline;
+import javafx.beans.value.ChangeListener;
+import javafx.css.PseudoClass;
 import javafx.scene.Node;
-import javafx.scene.control.Button;
+import javafx.scene.control.MenuItem;
 import javafx.util.Duration;
+import javafx.util.Pair;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.config.ConfigurableBeanFactory;
 import org.springframework.context.annotation.Scope;
 import org.springframework.stereotype.Component;
 import org.springframework.util.Assert;
 
-import java.time.OffsetDateTime;
+import static com.faforever.client.replay.LiveReplayService.LiveReplayAction.NOTIFY_ME;
+import static com.faforever.client.replay.LiveReplayService.LiveReplayAction.RUN;
 
 @Component
 @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@Slf4j
+@RequiredArgsConstructor
 public class WatchButtonController implements Controller<Node> {
+
+  public static final PseudoClass AVAILABLE_PSEUDO_CLASS = PseudoClass.getPseudoClass("available");
+  public static final PseudoClass TRACKABLE_PSEUDO_CLASS = PseudoClass.getPseudoClass("trackable");
+
+  private final LiveReplayService liveReplayService;
   private final ReplayService replayService;
-  private final ClientProperties clientProperties;
   private final TimeService timeService;
 
-  public Button watchButton;
+  public WatchLiveReplaySplitMenuButton watchButton;
+  public MenuItem notifyMeItem;
+  public MenuItem runReplayItem;
   private GameBean game;
   private final I18n i18n;
   private Timeline delayTimeline;
 
-  public WatchButtonController(ReplayService replayService, ClientProperties clientProperties, TimeService timeService, I18n i18n) {
-    this.replayService = replayService;
-    this.clientProperties = clientProperties;
-    this.timeService = timeService;
-    this.i18n = i18n;
-  }
+  private ChangeListener<Pair<Integer, LiveReplayAction>> trackingReplayPropertyListener;
 
   public void initialize() {
     delayTimeline = new Timeline(
         new KeyFrame(Duration.ZERO, event -> onFinished()),
         new KeyFrame(Duration.seconds(1))
     );
     delayTimeline.setCycleCount(Timeline.INDEFINITE);
-
-    watchButton.setDisable(true);
-    watchButton.setOnAction(event -> replayService.runLiveReplay(game.getId()));
+    watchButton.setIsCannotWatchSupplier(() -> !liveReplayService.canWatchReplay(game));
+    initializeMenuItemsByDefault();
   }
 
   public void setGame(GameBean game) {
-    this.game = game;
     Assert.notNull(game, "Game must not be null");
     Assert.notNull(game.getStartTime(), "The game's start must not be null: " + game);
-    if (canWatch()) {
+
+    this.game = game;
+    if (liveReplayService.canWatchReplay(game)) {
       allowWatch();
     } else {
+      initializeListeners();
```

Comment:
It looks like the listeners can be created in the initialize function. I don't see a reason why it has to be recreated each time a game is set.